### PR TITLE
Fix #38

### LINKED ISSUE #38: Most setters should update internal color value and return this
**Describe the bug**
Upon checking from the [original](https://github.com/bgrins/TinyColor/blob/master/tinycolor.js#L214-L224) code, setters update the internal color value and return its own reference.
Yet, most of the setters of the port of this library do not update and return a new `TinyColor`. See [here](https://github.com/TinyCommunity/tinycolor2/blob/master/lib/src/tinycolor.dart#L71-L76) for example.

This is an inconsistent behavior that should be fixed. (`.setAlpha()` & `.setOpacity()` is behaving correctly.)

**Expected behavior**
After calling any setters, `.color` property should be updated and return the updated `this` instead of creating another `TinyColor`.

### PR DESCRIPTION
**Pull request by calvintam236**

### CODE REVIEW

**Comment by calvintam236** [comment]
Blocked by #18

File: lib/src/tinycolor.dart [hunk 1]
```diff
@@ -125,38 +127,42 @@ class TinyColor {
...
     final p = amount / 100.0;
-    final color = Color.fromARGB(
+    _color = Color.fromARGB(
       ((toColor.alpha - _color.alpha) * p + _color.alpha).round(),
       ((toColor.red - _color.red) * p + _color.red).round(),
       ((toColor.green - _color.green) * p + _color.green).round(),
       ((toColor.blue - _color.blue) * p + _color.blue).round(),
     );
-    return TinyColor.fromColor(color);
+    return this;
```
Comment (calvintam236) [inline lib/src/tinycolor.dart:158]: @WieFel Maybe I am not thinking straight yet. I am confused why the return value has changed as it shouldn’t.

Comment (WieFel) [reply lib/src/tinycolor.dart:158]: @calvintam236 does it only happen with `mix` or also with other functions?

Comment (calvintam236) [reply lib/src/tinycolor.dart:158]: https://github.com/TinyCommunity/tinycolor2/runs/6042102828

`.shade()` failed due to using `.mix()`.

File: lib/src/tinycolor.dart [hunk 2]
```diff
@@ -103,13 +104,14 @@ class TinyColor {
...
   }
 
   TinyColor darken([int amount = 10]) {
     final hsl = toHSLColor();
     final lightness = clamp01(hsl.lightness - amount / 100);
-    return TinyColor.fromHSL(hsl.withLightness(lightness));
+    _color = hsl.withLightness(lightness).toColor();
+    return this;
```
Comment (calvintam236) [inline lib/src/tinycolor.dart:114]: @WieFel I need to check this why it failed in test. Any clue?

File: lib/src/tinycolor.dart (hunk 1 above)
Comment (calvintam236) [reply lib/src/tinycolor.dart:158]: @WieFel I am thinking could this part causing that?
See https://github.com/bgrins/TinyColor/blob/master/tinycolor.js#L38-L44.

Comment (WieFel) [reply lib/src/tinycolor.dart:158]: Yeah, looks indeed a bit strange.
I'm sorry for being so unresponsive currently... Very busy...

**Comment by calvintam236** [comment]
@WieFel looks like #58 fixed the issues..  probably due to how the `color` was changed before another test case.

**Review by WieFel — APPROVED** [review]

**Pull request merged**